Add more shape functions for mobilenet

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59051 Add peephole for len(x.size())
* **#58932 Add more shape functions for mobilenet**
* #58890 Switch symbolic shape registry to operator map'
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet

This adds all the operators necessary for mobilenet. I kind of wanted to get these landed to unblock @ZolotukhinM, but I'm happy to split these up into multiple PRs if it makes reviewing easier. In terms of testing, i'm going to add an automated shape analysis OpInfo test.

Differential Revision: [D28727246](https://our.internmc.facebook.com/intern/diff/D28727246)

eellison committed:
Add more functions

[ghstack-poisoned]

eellison committed:
Update on "Add more shape functions for mobilenet"




[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Thanks a lot for this stuff! Assuming the logic is not new and more tests will follow, this PR looks good to me to land!

eellison committed:
Update on "Add more shape functions for mobilenet"


This adds all the operators necessary for mobilenet. I kind of wanted to get these landed to unblock @ZolotukhinM, but I'm happy to split these up into multiple PRs if it makes reviewing easier. In terms of testing, i'm going to add an automated shape analysis OpInfo test.

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727246).

eellison committed:
Update on "Add more shape functions for mobilenet"


This adds all the operators necessary for mobilenet. I kind of wanted to get these landed to unblock @ZolotukhinM, but I'm happy to split these up into multiple PRs if it makes reviewing easier. In terms of testing, i'm going to add an automated shape analysis OpInfo test.

Differential Revision: [D28727246](https://our.internmc.facebook.com/intern/diff/D28727246)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727246).

eellison commented on test/jit/test_symbolic_shape_analysis.py:
```diff
@@ -96,6 +104,16 @@ def apply(fn):
             torch._C._jit_pass_propagate_shapes_on_graph(t.graph)
             self.assertEqual(next(t.graph.outputs()).type().symbolic_sizes(), [2, 2])
 
+    @skipIfNoTorchVision
```
this test is weirdly failing on just a couple platforms, and it doesnt actually test shape engine because output will always be [1, 1000] regardless of inputs so im removing it

eellison replied:
although it was useful to verify by looking at graph

eellison committed:
Update on "Add more shape functions for mobilenet"


This adds all the operators necessary for mobilenet. I kind of wanted to get these landed to unblock @ZolotukhinM, but I'm happy to split these up into multiple PRs if it makes reviewing easier. In terms of testing, i'm going to add an automated shape analysis OpInfo test.

Differential Revision: [D28727246](https://our.internmc.facebook.com/intern/diff/D28727246)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727246).
